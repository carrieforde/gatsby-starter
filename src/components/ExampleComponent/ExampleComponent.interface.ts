export interface ExampleComponentProps {
  testId?: string;
  name?: string;
}
