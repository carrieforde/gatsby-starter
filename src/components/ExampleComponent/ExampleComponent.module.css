.exampleComponent {
  color: rebeccapurple;
  font-family: sans-serif;
  font-weight: 600;
}
