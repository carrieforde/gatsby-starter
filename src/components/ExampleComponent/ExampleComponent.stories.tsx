import React from "react";
import ExampleComponent from "./ExampleComponent";
import { ExampleComponentProps } from "./ExampleComponent.interface";
import { Story, Meta } from "@storybook/react";

export default {
  title: "ExampleComponent",
  component: ExampleComponent,
} as Meta;

const Template: Story<ExampleComponentProps> = (args) => (
  <ExampleComponent {...args} />
);

export const DefaultExample = Template.bind({});
export const NameExample = Template.bind({});

NameExample.args = {
  name: "Carrie",
};
