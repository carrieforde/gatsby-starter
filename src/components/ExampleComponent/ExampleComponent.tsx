import React from "react";
import { ExampleComponentProps } from "./ExampleComponent.interface";
import { exampleComponent } from "./ExampleComponent.module.css";

const ExampleComponent: React.FC<ExampleComponentProps> = ({
  testId,
  name,
}) => (
  <p data-testid={testId} className={exampleComponent}>
    Hello {name ?? "world"}!
  </p>
);

export default ExampleComponent;
