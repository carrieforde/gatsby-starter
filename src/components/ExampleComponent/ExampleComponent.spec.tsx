import React from "react";
import ExampleComponent from "./ExampleComponent";
import { render } from "@testing-library/react";
import { DefaultExample, NameExample } from "./ExampleComponent.stories";

describe("ExampleComponent", () => {
  it("should render the ExampleComponent component", () => {
    const { container } = render(<ExampleComponent />);
    expect(container).toMatchSnapshot();
  });

  it("should print 'Hello world'", () => {
    const { container } = render(<DefaultExample {...DefaultExample.args} />);
    expect(container.firstElementChild).toHaveTextContent(/hello world/i);
  });

  it("should print 'Hello Carrie'", () => {
    const { container } = render(<NameExample {...NameExample.args} />);
    expect(container.firstElementChild).toHaveTextContent(/hello carrie/i);
  });
});
