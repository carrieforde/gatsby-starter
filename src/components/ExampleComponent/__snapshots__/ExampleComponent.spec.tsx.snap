// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent should render the ExampleComponent component 1`] = `
<div>
  <p
    class="exampleComponent"
  >
    Hello 
    world
    !
  </p>
</div>
`;
